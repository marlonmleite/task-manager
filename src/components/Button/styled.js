import styled from 'styled-components'

export const ButtonStyled = styled.button`
  padding: 0.4375rem 0.75rem;
  line-height: 1.25rem;
  font-size: 0.875rem;
  background-color: ${({ theme, color }) => theme.colors[color]};
  color: ${({ theme, color }) => theme.button.text[color]};
  display: inline-block;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  user-select: none;
  border: 1px solid ${({ theme, color }) => theme.colors[color]};
  border-radius: 2px;
  position: relative;

  :focus,
  :hover {
    text-decoration: none;
    background-image: linear-gradient(transparent, rgba(0,0,0,.05) 40%, rgba(0,0,0,.1));
  }

  :focus {
    outline: none;
  }

  .disabled,
  :disabled {
    opacity: .65;
    box-shadow: none;
  }

  :not([disabled]):not(.disabled) {
    cursor: pointer;
  }

  ${({ loading }) => loading && `
    > span {
      visibility: hidden;
    }
  `};
`

export const LoadingContainer = styled.div`
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
`
