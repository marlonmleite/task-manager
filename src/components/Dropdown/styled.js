import styled, { css } from 'styled-components'

export const Dropdown = styled.div`
  position: relative;

  > button {
    width: 100%;
  }
`

export const DropdownContent = styled.div`
  position: absolute;
  min-width: 200px;
  color: #89898a;
  background-color: #fff;
  border-radius: 2px;
  margin-top: 2px;
  padding: 3px 5px;
  top: 100%;
  left: 0;
  z-index: 99;
  display: flex;
  flex-direction: column;
  box-shadow: 0 8px 10px 1px rgba(0,0,0,0.14), 0 3px 14px 2px rgba(0,0,0,0.12), 0 5px 5px -3px rgba(0,0,0,0.2);

  ${({ right }) => right && css`
    left: auto;
    right: 0;
  `};
`

export const DropdownButton = styled.div`
  display: flex;
  user-select: none;
  align-items: center;
  justify-content: space-between;

  > span {
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
  }
`

export const DropdownOption = styled.div`
  width: 100%;
  position: relative;
  padding: 0 1.25rem 0 1.5rem;
  box-sizing: border-box;
  color: #222;
  height: 36px;
  line-height: 36px;
  font-size: 14px;

  :hover {
    background-color: #eee;
    cursor: pointer;
  }

  ${({ selected }) => selected && css`
    :before {
      content: '';
      border-left: 2px solid #666;
      border-bottom: 2px solid #666;
      width: 10px;
      height: 6px;
      margin-top: -7px;
      position: absolute;
      left: 5px;
      top: 50%;
      transform: rotate(-45deg);
    }
  `};
`

export const CaretStyled = styled.div`
  margin-left: 10px;
  width: 0px;
  height: 0px;
  border-left: 5px solid transparent;
  border-right: 5px solid transparent;
  border-top: 5px solid ${({ color, theme }) => theme.button.text[color]};
`
