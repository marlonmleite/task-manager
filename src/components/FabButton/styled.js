import styled from 'styled-components'
import Button from 'components/Button'

const FabButton = styled(Button)`
  padding: 1rem;
  border-radius: 100%;
  width: 56px;
  height: 56px;
  position: fixed;
  right: 20px;
  bottom: 25px;
  box-shadow: 0 0 4px rgba(0,0,0,.04), 0 4px 8px rgba(0,0,0,.08);
`

export default FabButton
