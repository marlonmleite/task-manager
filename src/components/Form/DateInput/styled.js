import styled from 'styled-components'

const Date = styled.div`
  .react-datepicker {
    border-radius: 2px;
    font-family: inherit;
  }
`

export default Date
