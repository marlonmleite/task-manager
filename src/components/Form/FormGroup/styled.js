import styled, { css } from 'styled-components'

const form = {
  labelOffset: '25px',
  paddingTop: '5px',
}

const topLabel = css`
  top: calc(${form.labelOffset} - (${form.paddingTop} + 10px));
  font-size: 12px;
`

export const FormGroupStyled = styled.div`
  width: 100%;
  font-size: 15px;
  position: relative;
  display: inline-flex;
  flex-direction: column; 
  color: ${({ theme }) => theme.form.label};
  padding: ${form.labelOffset} 0 10px 0;

  label {
    position: absolute;
    display: block;
    top: calc(${form.labelOffset} + ${form.paddingTop});
    left: 0px;
    transition: 0.2s ease all;
    pointer-events: none;
    cursor: text;
  }

  input {
    :focus ~ label,
    :not([value=""]) ~ label {
      ${topLabel};
    }
  }

  select,
  div {
    :focus ~ label,
    :not(.empty) ~ label {
      ${topLabel};
    }
  }

  input,
  select {
    :focus ~ label {
      color: ${({ theme }) => theme.focus};
    }
  }

  ${({ theme, invalid }) => invalid && css`
    input,
    select,
    input:focus,
    select:focus {
      border-color: ${theme.colors.danger};
    }

    input:focus ~ label,
    select:focus ~ label {
      color: inherit;
    }
  `};
`

export const ErrorStyled = styled.span`
  color: ${({ theme }) => theme.colors.danger};
  font-size: 13px;
`
