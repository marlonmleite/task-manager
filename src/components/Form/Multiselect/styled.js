import Select from 'react-select'
import styled from 'styled-components'

const SelectStyled = styled(Select)`
  .multiselect {
    &__control {
      border-color: #e5e5e5;
      border-style: solid;
      border-width: 0 0 1px 0;
      border-radius: 0;
      background-color: transparent;
      min-height: 38px;

      &--is-focused {
        box-shadow: none;
        border-color: ${({ theme }) => theme.focus};
      }
    }

    &__value-container {
      padding: 5px 5px 5px 0;
    }

    &__menu {
      border-radius: 2px;
    }
  }
`

export default SelectStyled
