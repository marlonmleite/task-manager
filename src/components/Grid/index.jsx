import PropTypes from 'prop-types'
import React, { Component } from 'react'
import map from 'lodash/map'
import { Grid as GridStyled, GridItem, Remove, Finish } from './styled'

class Grid extends Component {

  render() {
    const { items, render } = this.props

    return (
      <GridStyled>
        {map(items, (item, index) => {
          const key = `grid-${index}`

          return (
            <GridItem key={key}>
              {render(item, index)}
              <div className="grid-actions">
                <Remove />
                <Finish />
              </div>
            </GridItem>
          )
        })}
      </GridStyled>
    )
  }

}

Grid.propTypes = {
  items: PropTypes.array.isRequired,
  render: PropTypes.func.isRequired,
}

export default Grid
