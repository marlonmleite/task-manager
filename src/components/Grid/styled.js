import styled from 'styled-components'

export const Grid = styled.div`
  display: flex;
  flex-direction: column;
`

export const GridItem = styled.div`
  padding: 6px 0;
  border-bottom: 1px solid #e0e0e0;
`
