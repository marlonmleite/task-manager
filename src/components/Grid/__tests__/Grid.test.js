import React from 'react'
import { shallow } from 'enzyme'
import Grid from '../'
import { Edit, Remove, Finish } from '../styled'

describe('Grid', () => {

  const items = [{ id: 1, completed: false }, { id: 2, completed: true }]
  const singleItems = [items[0]]

  it('should render', () => {
    const grid = shallow(<Grid
      items={items}
      onRemove={jest.fn()}
      onEdit={jest.fn()}
      render={jest.fn()}
    />)

    expect(grid).toMatchSnapshot()
  })

  it('should item render prop call', () => {
    const render = jest.fn()
    const item = items[0]

    shallow(<Grid
      items={singleItems}
      onRemove={jest.fn()}
      onEdit={jest.fn()}
      render={render}
    />)

    expect(render).toHaveBeenCalledWith(item, 0)
  })

  it('should handle edit', () => {
    const onEdit = jest.fn()
    const grid = shallow(<Grid
      items={singleItems}
      onRemove={jest.fn()}
      onEdit={onEdit}
      render={jest.fn()}
    />)

    grid.find(Edit).simulate('click')

    expect(onEdit).toHaveBeenCalled()
  })

  it('should handle remove', () => {
    const onRemove = jest.fn()
    const grid = shallow(<Grid
      items={singleItems}
      onRemove={onRemove}
      onEdit={jest.fn()}
      render={jest.fn()}
    />)

    grid.find(Remove).simulate('click')

    expect(onRemove).toHaveBeenCalled()
  })

  it('should handle finish', () => {
    const onFinish = jest.fn()
    const grid = shallow(<Grid
      items={singleItems}
      onRemove={jest.fn()}
      onEdit={jest.fn()}
      onFinish={onFinish}
      render={jest.fn()}
    />)

    grid.find(Finish).simulate('click')

    expect(onFinish).toHaveBeenCalled()
  })


})
