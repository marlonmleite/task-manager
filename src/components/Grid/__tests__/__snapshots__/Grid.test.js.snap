// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid should render 1`] = `
<styled.div>
  <styled.div
    key="grid-0"
  >
    <div
      className="grid-actions"
    >
      <styled.jest-stub
        onClick={[Function]}
        role="button"
      />
      <styled.jest-stub
        onClick={[Function]}
        role="button"
      />
      <styled.jest-stub
        onClick={[Function]}
        role="button"
      />
    </div>
  </styled.div>
  <styled.div
    key="grid-1"
  >
    <div
      className="grid-actions"
    >
      <styled.jest-stub
        onClick={[Function]}
        role="button"
      />
      <styled.jest-stub
        onClick={[Function]}
        role="button"
      />
    </div>
  </styled.div>
</styled.div>
`;
